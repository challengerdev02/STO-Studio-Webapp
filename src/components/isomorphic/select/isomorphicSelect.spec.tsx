import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { IsomorphicSelect } from './index';

describe('Isomorphic select component', function () {
  afterEach(cleanup);

  const setup = () => {
    const utils = render(<IsomorphicSelect />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should check that the Isomorpic select component displays', () => {
    const { getByTestId } = setup();

    expect(getByTestId('isomorphic-select')).toBeInTheDocument();
  });
});
