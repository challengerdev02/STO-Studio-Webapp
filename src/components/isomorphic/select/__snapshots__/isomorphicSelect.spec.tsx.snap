// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Isomorphic select component should match with snapshot 1`] = `
<div>
  <div
    class="ant-select ant-select-borderless sc-bdvvtL kLdNxV ant-select-single ant-select-show-arrow"
    data-testid="isomorphic-select"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-item"
        title="Blockchain"
      >
        Blockchain
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <i
        class="mc-arrow-down-simple-line mc-2x down-arrow"
      />
    </span>
  </div>
</div>
`;
